diff-containers/README.md

```diff
@@ -1 +1,233 @@
-# diff-containers
\ No newline at end of file
+# diff-containers
+
+This package implements difference datatypes for container types. Currently,
+there is only one such implemented difference datatype for `Data.Map.Strict`.
+
+## Differences on `Data.Map`s
+
+A difference datatype for maps is defined and implemented in the
+`Data.Map.Diff.Strict` module. A key property of our definition of diffs is that
+diffs can not only combined/summed, they can also be inverted and summed
+together, which facilitates *subtraction* of diffs. Our definition of diffs is
+not only a `Semigroup` and a `Monoid`, it is also a `Group`.
```

Comment:
Hmm it may have been a confusing statement on my part, but a diff does have an identity element. It's diff entries that do not have any instances, because we don't need them to define the instances for diff histories and diffs themselves